src/engines/julia/update_rules/cvi.jl

```diff
@@ -0,0 +1,356 @@
+export ruleSPCVIOutNFactorNode, ruleSPCVIIn1Factor, ruleSPCVIOutNFactorNodeX, ruleSPCVIInFactorX, ruleMCVIFactorX
+
+function ruleSPCVIIn1Factor(node_id::Symbol,
+                            msg_out::Message{<:FactorFunction, <:VariateType},
+                            msg_in::Message{<:FactorNode, <:VariateType})
+
+    thenode = currentGraph().nodes[node_id]
```

Comment:
This is dangerous. What if active graph changes in a global environment?
But I understand you can't pass a node object to this function. Maybe @ThijsvdLaar can see a better solution here.